schema/schema.go

```diff
@@ -299,6 +299,14 @@ func ParseWithSpecialTableName(dest interface{}, cacheStore *sync.Map, namer Nam
 				field.HasDefaultValue = true
 				field.AutoIncrement = true
 			}
+		case String:
+			if _, ok := field.TagSettings["PRIMARYKEY"]; !ok {
+				if !field.HasDefaultValue || field.DefaultValueInterface != nil {
+					schema.FieldsWithDefaultDBValue = append(schema.FieldsWithDefaultDBValue, field)
+				}
+
+				field.HasDefaultValue = true
```

Comment:
I don't think we should allow default value on primarykeys unless there is a function to autoincrement in the database

Reply:
why not? it's okay that an insert fails if you don't provide a primary key and didn't setup a default mechanism. It's not okay to pass a zero value instead, as it would only prevent it for failing the first time and failing after that for being non unique.

Reply:
I support this change. After all this behavior existed before. And we should think that DB may generate id. Why Gorm should care of id generation?